### DISCUSSION TITLE
Brown out detection in zephyr

### DISCUSSION THREAD

**Discussion by MaazSk** [zephyrproject-rtos/zephyr#80231] [Q&A] [answered]
Does zephyr provide support for brown out detection. are there any apis to enable and disable the same? 

The only thing I am able to find close to [BOR is this.](https://docs.zephyrproject.org/apidoc/latest/group__hwinfo__interface.html#gac9dc9b22ba1fc551a3d68810155a640e) Can someone please help me how to enable and disable it for stm32h7. Or do I need to use HAL or something else?

**Comment by yashi** [comment]
Enable `CONFIG_HWINFO_STM32`?

I'm not STM expart but here is the driver: `drivers/hwinfo/hwinfo_stm32.c`

**Comment by FRASTM** [comment]
Yes, you have hwinfo_get_reset_cause() which gives you the reset cause
(for example, see samples/boards/st/power_mgmt/standby_shutdown/src/main.c )

**Reply by MaazSk** [reply]
I want to disable my brownout detection. Could you please guide me how to do it?

**Reply by FRASTM** [reply]
Not sure to catch what is your purpose. On a system reset, the stm32H7 mcu has positioned a flag   (in its RCC RSR status register)  indicating if brownout reset occured or any other reset source (watchdog, pin). If you want not to detect it,  just _do not read_ the flag (do not call the hwinfo_get_reset_cause or  clear it unconditionnaly with hwinfo_clear_reset_cause)
The hwinfo_get_reset_cause is a zephyr API, but there is none hwinfo API for enabling/disabling reset_cause()

**Reply by MaazSk** [reply]
Here is the thing I just want to disable the BOR for my application and I don't want the BOR to be triggered so is there a way to do it using zephyr apis or any other method?

**Reply by FRASTM** [reply]
The BOR signal  (bit BORRSTF of the RCC_RSR reset status reg )  is a  signal automatically  "Set by hardware when a BOR reset occurs"   that cannot be disabled.
Clearing is still possible by software by "writing the RMVF bit",  for example during the soc_early_init_hook(),  but what would be the benefit if you never read that flag ?

**Reply by MaazSk** [reply]
Let me rephrase the question again

In the image below
![image](https://github.com/user-attachments/assets/c8724afe-2081-48f0-9719-00152fbad73f)
We can set the BOR to OFF. So, I want to understand if there is any zephyr API to set the BOR to the above levels?
I know this can be done through option bytes in the stm32 cube programmer and also through ST HAL library but does zephyr provide any interface to do so?

**Comment by FRASTM** [comment, marked as answer]
Thank for clarification @MaazSk 
There is no zephyr API to change the option bytes in the stm32 flash as this is very specific to stm32 devices.
However there is a way to manipulate the Option Bytes through stm32Cube HAL driver layer. This is done as  example, in the PR https://github.com/zephyrproject-rtos/zephyr/pull/79985 :   define the OBInit.USERConfig and program with HAL_FLASHEx_OBProgram

**Reply by MaazSk** [reply]
Thank you! @FRASTM,
